src/main.rs

```diff
@@ -577,6 +577,25 @@ async fn start_ldk() {
 		Err(e) => println!("ERROR: errored reading channel peer info from disk: {:?}", e),
 	}
 
+	// Regularly broadcast our node_announcement. This is only required (or possible) if we have
+	// some public channels, and is only useful if we have public listen address(es) to announce.
+	// In a production environment, this should occur only after the announcement of new channels
+	// to avoid churn in the global network graph.
+	let chan_manager = Arc::clone(&channel_manager);
+	let network = args.network;
+	tokio::spawn(async move {
+		let mut interval = tokio::time::interval(Duration::from_secs(60));
+		loop {
+			interval.tick().await;
+			for chan_info in chan_manager.list_usable_channels() {
+				if chan_info.is_public && args.ldk_announced_listen_addr.is_some() {
```

Comment:
Should the entire `spawn` be dependent on `ldk_announced_listen_addr.is_some()` instead?